proxy/http/Http1ClientSession.cc

```diff
@@ -141,7 +141,7 @@ Http1ClientSession::new_connection(NetVConnection *new_vc, MIOBuffer *iobuf, IOB
   trans.mutex = mutex; // Share this mutex with the transaction
   in_destroy  = false;
 
-  TLSEarlyDataSupport *eds = dynamic_cast<TLSEarlyDataSupport *>(new_vc);
+  TLSEarlyDataSupport *eds = new_vc->get_service<TLSEarlyDataSupport>();
   if (eds != nullptr) {
```

Comment:
```suggestion
  if (TLSEarlyDataSupport *eds = new_vc->get_service<TLSEarlyDataSupport>()) {
```

We should be consistent about how we call `get_service`.